### ISSUE TITLE
plotly TriSurface seems broken

### ISSUE THREAD

**Issue by jimmytudeski** [holoviz/holoviews#4521]
#### ALL software version info
plotly, holoviews '1.13.3', firefox, jupyter notebook

#### Description of expected behavior and the observed behavior
I filed a problem on the [forum](https://discourse.holoviz.org/t/hover-on-plotly-trisurface-shows-wrong-values/831). The problem there was that the values, shown by the hover, were wrong. But now (with a new conda env installation) the plotly TriSurface seems completely broken.   The Surface example is working though.

#### Complete, minimal, self-contained example code that reproduces the issue

```
# jupyter notebook
import numpy as np
import holoviews as hv
hv.extension('plotly')
y,x = np.mgrid[-5:5, -5:5] * 0.2 
heights = np.sin(x**2+y**2) * 10

trisurface = hv.TriSurface((x.flat,y.flat,heights.flat)) 

trisurface.opts(height=500, width=500)
```

#### Stack traceback and/or browser JavaScript console output
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
~\Miniconda3\envs\holo\lib\site-packages\IPython\core\formatters.py in __call__(self, obj, include, exclude)
    968 
    969             if method is not None:
--> 970                 return method(include=include, exclude=exclude)
    971             return None
    972         else:

~\Miniconda3\envs\holo\lib\site-packages\holoviews\core\dimension.py in _repr_mimebundle_(self, include, exclude)
   1310         combined and returned.
   1311         """
-> 1312         return Store.render(self)
   1313 
   1314 

~\Miniconda3\envs\holo\lib\site-packages\holoviews\core\options.py in render(cls, obj)
   1393         data, metadata = {}, {}
   1394         for hook in hooks:
-> 1395             ret = hook(obj)
   1396             if ret is None:
   1397                 continue

~\Miniconda3\envs\holo\lib\site-packages\holoviews\ipython\display_hooks.py in pprint_display(obj)
    280     if not ip.display_formatter.formatters['text/plain'].pprint:
    281         return None
--> 282     return display(obj, raw_output=True)
    283 
    284 

~\Miniconda3\envs\holo\lib\site-packages\holoviews\ipython\display_hooks.py in display(obj, raw_output, **kwargs)
    250     elif isinstance(obj, (CompositeOverlay, ViewableElement)):
    251         with option_state(obj):
--> 252             output = element_display(obj)
    253     elif isinstance(obj, (Layout, NdLayout, AdjointLayout)):
    254         with option_state(obj):

~\Miniconda3\envs\holo\lib\site-packages\holoviews\ipython\display_hooks.py in wrapped(element)
    144         try:
    145             max_frames = OutputSettings.options['max_frames']
--> 146             mimebundle = fn(element, max_frames=max_frames)
    147             if mimebundle is None:
    148                 return {}, {}

~\Miniconda3\envs\holo\lib\site-packages\holoviews\ipython\display_hooks.py in element_display(element, max_frames)
    190         return None
    191 
--> 192     return render(element)
    193 
    194 

~\Miniconda3\envs\holo\lib\site-packages\holoviews\ipython\display_hooks.py in render(obj, **kwargs)
     66         renderer = renderer.instance(fig='png')
     67 
---> 68     return renderer.components(obj, **kwargs)
     69 
     70 

~\Miniconda3\envs\holo\lib\site-packages\holoviews\plotting\renderer.py in components(self, obj, fmt, comm, **kwargs)
    386                 doc = Document()
    387                 with config.set(embed=embed):
--> 388                     model = plot.layout._render_model(doc, comm)
    389                 if embed:
    390                     return render_model(model, comm)

~\Miniconda3\envs\holo\lib\site-packages\panel\viewable.py in _render_model(self, doc, comm)
    416         if comm is None:
    417             comm = state._comm_manager.get_server_comm()
--> 418         model = self.get_root(doc, comm)
    419 
    420         if config.embed:

~\Miniconda3\envs\holo\lib\site-packages\panel\viewable.py in get_root(self, doc, comm)
    645         """
    646         doc = doc or _curdoc()
--> 647         root = self._get_model(doc, comm=comm)
    648         self._preprocess(root)
    649         ref = root.ref['id']

~\Miniconda3\envs\holo\lib\site-packages\panel\layout.py in _get_model(self, doc, root, parent, comm)
    118         if root is None:
    119             root = model
--> 120         objects = self._get_objects(model, [], doc, root, comm)
    121         props = dict(self._init_properties(), objects=objects)
    122         model.update(**self._process_param_change(props))

~\Miniconda3\envs\holo\lib\site-packages\panel\layout.py in _get_objects(self, model, old_objects, doc, root, comm)
    108             else:
    109                 try:
--> 110                     child = pane._get_model(doc, root, model, comm)
    111                 except RerenderError:
    112                     return self._get_objects(model, current_objects[:i], doc, root, comm)

~\Miniconda3\envs\holo\lib\site-packages\panel\pane\holoviews.py in _get_model(self, doc, root, parent, comm)
    225             plot = self.object
    226         else:
--> 227             plot = self._render(doc, comm, root)
    228 
    229         plot.pane = self

~\Miniconda3\envs\holo\lib\site-packages\panel\pane\holoviews.py in _render(self, doc, comm, root)
    284             kwargs = {}
    285 
--> 286         return renderer.get_plot(self.object, **kwargs)
    287 
    288     def _cleanup(self, root):

~\Miniconda3\envs\holo\lib\site-packages\holoviews\plotting\renderer.py in get_plot(self_or_cls, obj, doc, renderer, comm, **kwargs)
    235             init_key = tuple(v if d is None else d for v, d in
    236                              zip(plot.keys[0], defaults))
--> 237             plot.update(init_key)
    238         else:
    239             plot = obj

~\Miniconda3\envs\holo\lib\site-packages\holoviews\plotting\plot.py in update(self, key)
    898     def update(self, key):
    899         if len(self) == 1 and ((key == 0) or (key == self.keys[0])) and not self.drawn:
--> 900             return self.initialize_plot()
    901         item = self.__getitem__(key)
    902         self.traverse(lambda x: setattr(x, '_updated', True))

~\Miniconda3\envs\holo\lib\site-packages\holoviews\plotting\plotly\element.py in initialize_plot(self, ranges)
    119         """
    120         # Get element key and ranges for frame
--> 121         fig = self.generate_plot(self.keys[-1], ranges)
    122         self.drawn = True
    123 

~\Miniconda3\envs\holo\lib\site-packages\holoviews\plotting\plotly\element.py in generate_plot(self, key, ranges, element)
    150 
    151         # Get data and options and merge them
--> 152         data = self.get_data(element, ranges, style)
    153         opts = self.graph_options(element, ranges, style)
    154 

~\Miniconda3\envs\holo\lib\site-packages\holoviews\plotting\plotly\chart3d.py in get_data(self, element, ranges, style)
    101         x, y, z = (element.dimension_values(i) for i in range(3))
    102         points2D = np.vstack([x, y]).T
--> 103         tri = Delaunay(points2D)
    104         simplices = tri.simplices
    105         return [dict(x=x, y=y, z=z, simplices=simplices)]

UnboundLocalError: local variable 'Delaunay' referenced before assignment

:TriSurface   [x,y,z]

Thanks

**Comment by philippjfr** [comment]
It should skip rendering if this import fails:

```python
try:
    from scipy.spatial import Delaunay
except:
    SkipRendering("SciPy not available, cannot plot TriSurface")
```    

The fix is simple though, just install scipy.

**Comment by jimmytudeski** [comment]
Thank you, it works again. Could you give a comment on the hover thing as well ? (The hover tool in the example above shows wrong values for x,y and z) Should I open a new issue for that?

**Comment by philippjfr** [comment]
Sorry I never replied, the hover issue seem like a Plotly issue. At least the actual coordinates passed to Plotly are correct.

**Comment by philippjfr** [comment]
The SkipRendering exception wasn't raised which is why you didn't see the error message. That was fixed in c3c01f73c. Please open a new issue for hover if it turns out it's not Plotly causing this.
